restructuredtext_lint/sphinx.py

```diff
@@ -0,0 +1,127 @@
+from __future__ import absolute_import
+
+import contextlib
+
+try:
+    import sphinx  # noqa
+
+    from sphinx.domains import c as c_domain
+    from sphinx.domains import cpp as cpp_domain
+    from sphinx.domains import javascript as javascript_domain
+    from sphinx.domains import python as python_domain
+    from sphinx.domains import std as std_domain
+
+    SPHINX_AVAILABLE = True
+except ImportError:
+    SPHINX_AVAILABLE = False
+
+from docutils.parsers import rst
+from docutils.parsers.rst import directives as rst_directives
+from docutils.parsers.rst import roles as rst_roles
+
+_base_sp_roles = tuple(['ctype'])
+_base_sp_directives = tuple(['autosummary',
+                             'centered', 'currentmodule', 'deprecated',
+                             'hlist', 'include', 'index',
+                             'literalinclude', 'no-code-block',
+                             'seealso', 'toctree', 'todo',
+                             'versionadded', 'versionchanged'])
+
+
+class _IgnoredDirective(rst.Directive):
+    """Stub for unknown directives.
+
+    Thanks rstcheck!
+    """
+
+    has_content = True
```

Comment:
For consistency, please don't leave a newline between the end of a docstring and the next significant line of code